packages/public/server/__tests-e2e__/login.ts

```diff
@@ -103,38 +100,68 @@ describe('missing login-data', () => {
   let loginPage: ElementHandle
 
   beforeAll(async () => {
-    await page.setViewport(viewports.desktop)
     loginPage = await goto('/auth/login')
   })
+
   test('missing password and missing user', async () => {
-    typeIntoLoginForm('', loginPage, '')
-    clickOnLoginButton(loginPage)
+    await typeIntoLoginFormAndClickLoginButton({
+      password: '',
+      page: loginPage,
+      user: '',
+    })
     expect(page.url()).toBe(testingServerUrl + pages.login.path)
   })
 
   test('missing password and correct user', async () => {
-    typeIntoLoginForm('', loginPage, 'login')
-    clickOnLoginButton(loginPage)
+    await typeIntoLoginFormAndClickLoginButton({
+      password: '',
+      page: loginPage,
+      user: 'login',
+    })
     expect(page.url()).toBe(testingServerUrl + pages.login.path)
   })
 
   test('correct password and missing user', async () => {
-    typeIntoLoginForm('123456', loginPage, '')
-    clickOnLoginButton(loginPage)
+    await typeIntoLoginFormAndClickLoginButton({
+      password: '123456',
+      page: loginPage,
+      user: '',
+    })
     expect(page.url()).toBe(testingServerUrl + pages.login.path)
   })
 })
 
-async function typeIntoLoginForm(
-  password: string,
-  page: ElementHandle,
+async function typeIntoLoginFormAndClickLoginButton(arg: {
+  page: ElementHandle
   user: string
-) {
-  const { inputUser, inputPassword } = pages.login.identifier
-  await getByPlaceholderText(page, inputUser).then((e) => e.type(user))
-  await getByPlaceholderText(page, inputPassword).then((e) => e.type(password))
+  password: string
+}) {
+  {
+    await typeIntoLoginForm(arg)
+    const logbutton = elements.getLoginButton(arg.page)
+    ;(await logbutton).click
+  }
+}
+
+async function typeIntoLoginFormAndWaitForNewPage(arg: {
+  page: ElementHandle
+  user: string
+  password: string
+}) {
+  {
+    await typeIntoLoginForm(arg)
+    const { buttonLogin } = pages.login.identifier
+    return await getByText(arg.page, 'Login').then(clickForNewPage)
```

Comment:
```suggestion
    return getByText(arg.page, 'Login').then(clickForNewPage)
```

It should also work without `await`. I have no clear opinion what is better (I just want to point out, that this is an alternative). You can decide what you want to use :smile: